vcGL/src/vcTextureHelper.cpp

```diff
@@ -319,22 +320,42 @@ udResult vcTexture_ConvertPixels(const void *pPixels, vcTextureFormat formatIn,
   return result;
 }
 
+struct vcTextureCallbackResult
+{
+  udFile *pFile;
+  const char *pFilename;
+  udResult result = udR_Success;
+};
+
+void vcTexture_StbCallback(void *context, void *data, int size)
+{
+  vcTextureCallbackResult *pData = (vcTextureCallbackResult *)context;
+  udResult writeResult = udFile_Write(pData->pFile, data, size);
+
+  if (pData->result == udR_Success)
+    pData->result = writeResult;
+}
+
 udResult vcTexture_SaveImage(vcTexture *pTexture, vcFramebuffer *pFramebuffer, const char *pFilename)
 {
   if (pTexture == nullptr || pFramebuffer == nullptr || pFilename == nullptr)
     return udR_InvalidParameter_;
 
   udResult result;
-  int outLen = 0;
-  unsigned char *pWriteData = nullptr;
   vcTextureFormat format;
   size_t pixelSize;
   uint8_t *pFrameBufferPixels = nullptr;
   uint8_t *pPixelsOut = nullptr;
+  int stbiResult = 1;
+  vcTextureCallbackResult callbackResult;
+  callbackResult.pFilename = pFilename;
+  callbackResult.pFile = udAllocType(udFile, 1, udAF_Zero);
```

Comment:
This isn't how you use `udFile`, `udFile_Open` allocates for you, that's why you pass in a pointer to the `udFile*` and not just the `udFile*`.